Tribler/Main/Utility/GuiDBHandler.py

```diff
@@ -79,7 +75,7 @@ def onSameThread(self, type):
         if type == "dbThread" or isInIOThread():
             return isInIOThread()
 
-        return threading.currentThread().getName().startswith('GUITaskQueue')
+        return False
```

Comment:
the IOThread is the twisted thread, but this return statement should check if we're in a twisted thread pool